lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,325 @@
+        self._source_vhd: Optional[PureWindowsPath] = None
+
+    def _get_hyperv_runbook(self) -> HypervPlatformSchema:
+        hyperv_runbook = self.runbook.get_extended_runbook(HypervPlatformSchema)
+        assert hyperv_runbook, "platform runbook cannot be empty"
+        return cast(HypervPlatformSchema, hyperv_runbook)
+
+    def _initialize_server_node(self) -> RemoteNode:
+        assert self._hyperv_runbook, "hyperv runbook cannot be empty"
+
+        if len(self._hyperv_runbook.servers) > 1:
+            self._log.info(
+                "Multiple servers are currently not supported. "
+                "Only the first server will be used."
+            )
+
+        server_runbook = self._hyperv_runbook.servers[0]
+        server_node = RemoteNode(
+            runbook=schema.Node(name="hyperv"),
+            index=-1,
+            logger_name="hyperv",
+            parent_logger=get_logger("hyperv"),
+        )
+
+        server_node.set_connection_info(
+            address=server_runbook.address,
+            username=server_runbook.username,
+            password=server_runbook.password,
+        )
+
+        return server_node
+
+    def _prepare_environment(self, environment: Environment, log: Logger) -> bool:
+        if not environment.runbook.nodes_requirement:
+            return True
+
+        if len(environment.runbook.nodes_requirement) > 1:
+            log.error("Multiple nodes are currently not supported.")
+            return False
+
+        nodes_capabilities = self._create_node_capabilities()
+
+        nodes_requirement = []
+        for node_space in environment.runbook.nodes_requirement:
+            if not node_space.check(nodes_capabilities):
+                return False
+
+            requirement = node_space.generate_min_capability(nodes_capabilities)
+            nodes_requirement.append(requirement)
+
+        if not self._check_host_capabilities(nodes_requirement, log):
+            return False
+
+        environment.runbook.nodes_requirement = nodes_requirement
+        return True
+
+    def _get_host_capabilities(self, log: Logger) -> _HostCapabilities:
+        host_cap = _HostCapabilities()
+
+        free_mem_bytes = self._server.tools[PowerShell].run_cmdlet(
+            "(Get-CimInstance -ClassName Win32_OperatingSystem).FreePhysicalMemory"
+        )
+        host_cap.free_memory_mib = int(free_mem_bytes) // 1024
+
+        host_cap.free_memory_mib -= 2048  # reserve 2 GiB for host
+
+        lp_count = self._server.tools[PowerShell].run_cmdlet(
+            "(Get-CimInstance Win32_ComputerSystem).NumberOfLogicalProcessors"
+        )
+        host_cap.core_count = int(lp_count)
+
+        log.debug(
+            f"Host capabilities: {host_cap.core_count} cores, "
+            f"{host_cap.free_memory_mib} MiB free memory"
+        )
+
+        return host_cap
+
+    # Check that the VM requirements can be fulfilled by the host.
+    def _check_host_capabilities(
```

Comment:
The method name is too general, call it like `_is_host_resources_enough`.

Reply:
Done.